arrows/core/video_input_arf.cxx

```diff
@@ -0,0 +1,558 @@
+}
+
+// ------------------------------------------------------------------
+bool
+video_input_arf
+::check_configuration( VITAL_UNUSED vital::config_block_sptr config ) const
+{
+  return true;
+}
+
+// ------------------------------------------------------------------
+void
+video_input_arf
+::open( std::string filename)
+{
+  typedef kwiversys::SystemTools ST;
+
+  // close the video in case already open
+  this->close();
+
+  d->file = fopen(filename.c_str(), "rb");
+  if (d->file == nullptr)
+  {
+    VITAL_THROW(kv::invalid_file, filename, "Could not open file");
+  }
+
+  unsigned char header[32];
+  fread(header, 4, 8, d->file);
+  uint32_t* iheader = (uint32_t*)&header;
+  uint32_t magic = d->byteswap(iheader[0]);
+  if (magic != 3149642413)
+  {
+    VITAL_THROW(kv::invalid_file, filename, "Is not an ARF file");
+  }
+
+  d->version = d->byteswap(iheader[1]);
+  if (d->version != 2)
+  {
+    VITAL_THROW(kv::invalid_file, filename, "Unsupported ARF version");
+  }
+
+  d->rows = d->byteswap(iheader[2]);
+  d->cols = d->byteswap(iheader[3]);
+  uint32_t imgtype = d->byteswap(iheader[4]);
+  d->num_frames = d->byteswap(iheader[5]);
+  d->offset = d->byteswap(iheader[6]);
+  uint32_t flags = d->byteswap(iheader[7]);
+
+  bool arf_info = (flags & (2 ^ 0)) != 0;
+  bool arf_colormap = (flags & (2 ^ 1)) != 0;
+  bool arf_comment = (flags & (2 ^ 2)) != 0;
+  bool arf_multiband = (flags & (2 ^ 3)) != 0;
+  bool arf_framedata = (flags & (2 ^ 4)) != 0;
+  bool arf_extra = (flags & (2 ^ 5 - 1)) != 0;
+  if (arf_extra)
+    std::cout << "Additional unsupported/unimplemented ARF flags exist, read may fail\n";
+
+  if (flags == 1 && d->offset == 288)
+  {
+    std::cout << "header looks incorrect, assuming ARF_COMMENT instead of ARF_INFO\n";
+    arf_info = false;
+    arf_comment = true;
+  }
+
+  if (arf_info)
+  {
+    // TODO support this! (need example)
+    //fread(&image_source, 4, 1, d->file);
+    //fread(&start_x, 4, 1, d->file);
+    //fread(&start_y, 4, 1, d->file);
+    //fread(&nud->avg, 4, 1, d->file);
+    //fread(&capture_rate, 4, 1, d->file);
+    //fread(&capture_time, 4, 6, d->file);
+    //capture_loc = readString(32);
+    //sensor_name = readString(32);
+    //digitizer = readString(32);
+    //fread(&sensor_hfov, 1, 1, d->file);
+    //fread(&sensor_vfov, 1, 1, d->file);
+    //fread(&samples_per_dwell, 4, 1, d->file);
+  }
+
+  unsigned char color_map[COLOR_MAP_SIZE];
+  if (arf_colormap)
+  {
+    // Not sure what to do with this....
+    fread(&color_map, 1, COLOR_MAP_SIZE, d->file);
+  }
+
+  if (arf_comment)
+  {
+    //comment = readString(256);
+  }
+
+  if (arf_multiband)
+  {
+    // TBD
+  }
+
+  if (arf_framedata)
+  {
+    uint32_t footer_flags;
+    fread(&footer_flags, 4, 1, d->file);
+    d->frame_info = (footer_flags & (2 ^ 0)) != 0;
+    d->frame_pad = 2 * 4 + 2 * 4 + 6 * 4;//40 bytes
+  }
+
+  switch (imgtype)
+  {
+  case 0:
+    d->bpp = 1;//uint8
+    d->px.num_bytes = d->bpp;
```

Comment:
it looks like setting `num_bytes` is done the same way in all cases, so you could pull this out of the switch.